### PR TITLE
Server disable schematic option (fix #2382)

### PR DESCRIPTION
**Pull request by J-VdS**
* removed `copyAllowed`
* create a popup when trying to build a schematic
* add map rule

### CODE REVIEW

File: server/src/mindustry/server/ServerControl.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public ServerControl(String[] args){
             "bans", "",
             "admins", "",
             "shufflemode", "custom",
-            "globalrules", "{reactorExplosions: false}"
+            "globalrules", "{reactorExplosions: false}" //schematicAllowed: true} //is not working
```
Comment (Anuken) [inline server/src/mindustry/server/ServerControl.java:64]: It's true by default, why add it here in the first place?

File: core/src/mindustry/input/MobileInput.java [hunk 2]
```diff
@@ -488,7 +488,7 @@ public boolean touchUp(int screenX, int screenY, int pointer, KeyCode button){
             selectRequests.clear();
             lastSchematic = schematics.create(lineStartX, lineStartY, lastLineX, lastLineY);
             useSchematic(lastSchematic);
-            if(selectRequests.isEmpty()){
+            if (selectRequests.isEmpty()) {
```
Comment (Anuken) [inline core/src/mindustry/input/MobileInput.java:491]: Why?

File: core/src/mindustry/input/DesktopInput.java [hunk 3]
```diff
@@ -344,7 +344,7 @@ public void buildPlacementUI(Table table){
 
         table.button(Icon.paste, Styles.clearPartiali, () -> {
             ui.schematics.show();
-        }).tooltip("@schematics");
+        }).tooltip("Schematics");
```
Comment (Anuken) [inline core/src/mindustry/input/DesktopInput.java:347]: Why?

File: core/src/mindustry/input/DesktopInput.java [hunk 4]
```diff
@@ -416,7 +416,7 @@ void pollInput(){
         if(Core.input.keyRelease(Binding.schematic_select) && !Core.scene.hasKeyboard()){
             lastSchematic = schematics.create(schemX, schemY, rawCursorX, rawCursorY);
             useSchematic(lastSchematic);
-            if(selectRequests.isEmpty()){
+            if (selectRequests.isEmpty()) {
```
Comment (Anuken) [inline core/src/mindustry/input/DesktopInput.java:419]: Why?

File: core/assets/bundles/bundle.properties [hunk 5]
```diff
@@ -830,6 +832,7 @@ rules.lighting = Lighting
 rules.ambientlight = Ambient Light
 rules.solarpowermultiplier = Solar Power Multiplier
 
+
```
Comment (Anuken) [inline core/assets/bundles/bundle.properties:835]: Extra random indent

File: core/assets/bundles/bundle.properties [hunk 6]
```diff
@@ -803,6 +804,7 @@ mode.custom = Custom Rules
 
 rules.infiniteresources = Infinite Resources
 rules.reactorexplosions = Reactor Explosions
+rules.schematic = Schematics allowed
```
Comment (Anuken) [inline core/assets/bundles/bundle.properties:807]: "A" should be capitalized

File: core/src/mindustry/ui/dialogs/SchematicsDialog.java [hunk 7]
```diff
@@ -64,8 +65,12 @@ void setup(){
 
             t.update(() -> {
                 if(Core.input.keyTap(Binding.chat) && Core.scene.getKeyboardFocus() == searchField && firstSchematic != null){
-                    control.input.useSchematic(firstSchematic);
-                    hide();
+                    if(!Vars.state.rules.schematicAllowed){
+                        ui.showInfo("@schematic.disabled");
+                    }else {
```
Comment (Anuken) [inline core/src/mindustry/ui/dialogs/SchematicsDialog.java:70]: random space

**Review by Anuken — CHANGES_REQUESTED** [review]
Can't comment on a deletion directly:

image

Do not touch this file.

**Comment by Ilya246** [comment]
Why disallow/allow schematics though? Disallowing schematics will only make people not play on the server that disabled them

File: core/src/mindustry/input/DesktopInput.java (hunk 3 above)
Comment (J-VdS) [reply core/src/mindustry/input/DesktopInput.java:347]: typo

File: core/src/mindustry/input/MobileInput.java (hunk 2 above)
Comment (J-VdS) [reply core/src/mindustry/input/MobileInput.java]: idk, I didn't touch that file

File: server/src/mindustry/server/ServerControl.java (hunk 1 above)
Comment (J-VdS) [reply server/src/mindustry/server/ServerControl.java]: I removed the comment but I couldn't find a way to add an extra argument.

**Comment by J-VdS** [comment]
> 
> 
> Why would you disallow schematics though? Disallowing schematics will only make people not play on the server that disabled them

This PR was originally made bc I was working on a plugin. And some serverowners don't like (huge) schematics. And it is an extra challenge imo. More info in #2382

**Comment by mse-k** [comment]
yea no can you not add this because if you give map makers the choice to disable them a ton of maps and servers will have it disabled thinking it will help but in reality it just makes nobody want to play there, this is the worst idea ive ever seen and even worse is that you actually wasted time coding it

**Comment by J-VdS** [comment]
I made this bc I was making a "speedbuilder" plugin...

**Comment by ritzip** [comment]
> in reality it just makes nobody want to play there
A lot of people don't play with schematics, me included; I personally think schematics ruin gameplay.

**Comment by Anuken** [comment]
image

Again, un-deleted this file. I cannot merge this until you do.

**Pull request merged**